apps/next-docs/content/components/IDE/index.mdx

```diff
@@ -0,0 +1,294 @@
+storybook: '/brand/storybook/?path=/story/components-ide--playground'
+---
+
+import {Label} from '@primer/react'
+
+import {IDEVariantProp, IDEEditorSizeProp, IDEFileExtensionsList} from './react'
+
+```js
+import {IDE} from '@primer/react-brand'
+```
+
+## Examples
+
+### Default
+
+<Note variant="warning">
+
+The `IDE` does not automatically apply syntax highlighting to code snippets. Usage of this component in GitHub.com is expected to use the proprietary tokenizer and highlighter, the output of which can be passed directly into `IDE` as a parameter.
+
+</Note>
+
+```jsx live
+<IDE height={500}>
+  <IDE.Editor
+    showReplayButton={false}
+    files={[
+      {
+        name: 'index.js',
+        alternativeText: 'TypeScript sentiment analysis function with D3.js visualization.',
+        code: `import { fetch } from "fetch-h2";
+
+async function isPositive(text: string): Promise<boolean> {
+  const response = await fetch(...)
+  const drawScatterplot = (data, height, width) => {
+    const svg = d3.select("#scatterplot")
+      .data(data)
+      .attr("cx", d => d.x)
+      .attr("cy", d => d.y)
+      .attr("r", 5)
+  }
+  const json = await response.json();
+  return json.label === "pos";
+}`,
+      },
+      {
+        name: 'parse_expenses.py',
+        alternativeText:
+          'Python function that converts financial transaction strings into structured data, parsing dates, amounts, and currencies while skipping comments',
+        code: `import datetime
+
+def parse_expenses(expenses_string):
+Ignore lines starting with #.
+Parse the date using datetime.
+Example expenses_string:
+2016-01-02 -34.01 USD
+2016-01-03 2.59 DKK
+2016-01-03 -2.72 EUR
+"""
+expenses = []
+for line in expenses_string.splitlines():
+if line.startswith("#"):
+continue
+date, value, currency = line.split(" ")
+expenses.append((datetime.datetime.strptime(date, "%Y-%m-%d"),
+          float(value),
+          currency))
+return expenses`,
+      },
+    ]}
+  />
+</IDE>
+```
+
+### Animations
+
+Line-by-line animations can be enabled using pre-formatted prop values. A replay button is enabled by default.
+
+```jsx live
+<IDE height={500}>
+  <IDE.Editor
+    files={[
+      {
+        name: 'index.js',
+        alternativeText: 'TypeScript sentiment analysis function with D3.js visualization.',
+
+        code: `import { fetch } from "fetch-h2";
+
+async function isPositive(text: string): Promise<boolean> {
+  const response = await fetch(...)
+  const drawScatterplot = (data, height, width) => {
+    const svg = d3.select("#scatterplot")
+      .data(data)
+      .attr("cx", d => d.x)
+      .attr("cy", d => d.y)
+      .attr("r", 5)
+  }
+  const json = await response.json();
+  return json.label === "pos";
+}`
+          .split('\n')
+          .map(line => line || ' '),
+      },
+    ]}
+  />
+</IDE>
+```
+
+### GitHub Copilot suggestions
+
+Simulate GitHub Copilot suggestions using the `suggestedLineStart` property.
+
+```jsx live
+<IDE height={500}>
```

Comment:
Some rendering issues here too, looks like we need to exclude default pre styling from previews. Could we catch this in our board if we don't fix here please.
<img width="585" alt="Screenshot 2025-04-23 at 12 45 37" src="https://github.com/user-attachments/assets/269f96af-da87-4e28-a733-4764bb0287e9" />